advanced_source/extend_dispatcher.rst

```diff
@@ -0,0 +1,295 @@
+for all backends. In those cases backend extenders can optionally override PyTorch Autograd layer by registering
+a kernel from torch::autograd::Function to the corresponding dispatch key (for example, AutogradPrivateUse1 if
+you're using PrivateUse1 for your backend):
+
+
+.. code-block:: cpp
+
+
+  class MyAddFunction : public torch::autograd::Function<MyAddFunction> {
+    public:
+    static Tensor forward(AutogradContext *ctx, torch::Tensor self, torch::Tensor other) {
+      at::AutoNonVariableTypeMode g;
+      return myadd(self, other);
+    }
+
+    static tensor_list backward(AutogradContext *ctx, tensor_list grad_outputs) {
+      auto grad_output = grad_outputs[0];
+      return {grad_output, grad_output};
+    }
+  };
+
+  Tensor myadd_autograd(const Tensor& self, const Tensor& other) {
+    return MyAddFunction::apply(self, other)[0];
+  }
+
+  // Register the autograd kernel to AutogradPrivateUse1
+  TORCH_LIBRARY_IMPL(aten, AutogradPrivateUse1, m) {
+    m.impl(<myadd_schema>, &myadd_autograd);
+  }
+
+  // Register the inference kernel to PrivateUse1
+  TORCH_LIBRARY_IMPL(aten, PrivateUse1, m) {
+    m.impl(<myadd_schema>, &myadd);
+  }
+
+
+
+With this trick you have full control over both training and inference behavior for ``my_add`` operator in your backend.
+Here's `an example <https://github.com/pytorch/xla/blob/master/torch_xla/csrc/aten_autograd_ops.h>`_ in the ``pytorch/xla`` repository.
+
+
+Custom operator support
+-----------------------
+
+Your new backend should work seamlessly with
+`customized operators extended in python <https://pytorch.org/docs/stable/notes/extending.html>`_
+without writing any new kernels as long as the customized operator is composable of existing
+PyTorch operators (which are already supported by your backend).
+
+For `custom operators extended in C++ <cpp_autograd>`_ they often come with a
+`backend specific C++ kernel implementation e.g. nms kernel in torchvsion <https://github.com/pytorch/vision/blob/master/torchvision/csrc/ops/cuda/nms_kernel.cu>`_
+as well as `a customized Python API e.g. torch.ops.torchvision.nms <https://github.com/pytorch/vision/blob/master/torchvision/csrc/ops/nms.cpp#L18>`_.
+To support these operators, backend extenders will need to write a C++ kernel for your backend and properly
+register them to the corresponding namespace in the dispatcher similar to supporting PyTorch native operators.
+Alternatively you could also add a customized API in your extension e.g ``torch_xla.core.functions.nms`` for
+these adhoc requests.
+
+JIT support
+-----------
+
+As we mentioned in `Registering a Dispatched Operator in C++ <dispatcher>`_, kernels registered through `m.impl()` API
+supports being called in both unboxed and boxed ways. In other words your customized backend can also work with our
+JIT tracing/scripting frontend like native CPU/CUDA backend.  You could potentially also write specialized optimization
```

Comment:
I'd change "like native CPU/CUDA backend" to something like "just like the in-tree backends like CPU or CUDA do", just for clarity